### PR TITLE
do not use hardcoded ns for ocs-prometheus-rules

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/storagecluster/prometheus.go [hunk 1]
```diff
@@ -42,20 +41,20 @@ func (r *StorageClusterReconciler) enablePrometheusRules(instance *ocsv1.Storage
...
+func getPrometheusRules(instance *ocsv1.StorageCluster) (*monitoringv1.PrometheusRule, error) {
 	rule := &monitoringv1.PrometheusRule{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       monitoringv1.PrometheusRuleKind,
 			APIVersion: monitoringv1.SchemeGroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      ruleName,
-			Namespace: ruleNamespace,
+			Namespace: instance.Namespace,
```
Comment (umangachapagain) [inline controllers/storagecluster/prometheus.go:52]: We should use the namespace of operator here.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/prometheus.go (hunk 1 above)
Comment (iamniting) [reply controllers/storagecluster/prometheus.go]: using operator ns.

File: controllers/storagecluster/prometheus.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func getPrometheusRules(isExternal bool) (*monitoringv1.PrometheusRule, error) {
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      ruleName,
-			Namespace: ruleNamespace,
+			Namespace: os.Getenv(util.OperatorNamespaceEnvVar),
```
Comment (umangachapagain) [inline controllers/storagecluster/prometheus.go:53]: Use `util.GetOperatorNamespace()` for safety. This could fail is env var is not set.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/prometheus.go (hunk 2 above)
Comment (iamniting) [reply controllers/storagecluster/prometheus.go:53]: We are already doing that in the main.go, So it should be safe. https://github.com/red-hat-storage/ocs-operator/blob/main/main.go#L120

**Review by umangachapagain — APPROVED** [review]

**Pull request merged**